ui/raidboss/data/07-dt/ultimate/futures_rewritten.ts

```diff
@@ -481,13 +681,895 @@ const triggerSet: TriggerSet<Data> = {
+      id: 'FRU P3 Ultimate Rel 3rd Fire/Stack',
+      type: 'GainsEffect',
+      netRegex: { effectId: '9A0' },
+      condition: (data, matches) => data.phase === 'p3-ur' && data.me === matches.target,
+      delaySeconds: 26,
+      alertText: (data, _matches, output) => {
+        const debuff = data.p3RelativityDebuff;
+        if (debuff === undefined)
+          return;
+
+        if (data.triggerSetConfig.ultimateRel !== 'yNorthDPSEast')
+          return output[debuff]!();
+
+        const dirStr = data.p3RelativityMyDirStr;
+        if (debuff !== 'longFire')
+          return output.yNorthStrat!({ debuff: output[debuff]!(), dir: output.middle!() });
+        return output.yNorthStrat!({ debuff: output.longFire!(), dir: dirStr });
+      },
+      outputStrings: {
+        yNorthStrat: p3UROutputStrings.yNorthStrat,
+        shortFire: p3UROutputStrings.stack,
+        mediumFire: p3UROutputStrings.stack,
+        longFire: p3UROutputStrings.fireSpread,
+        ice: p3UROutputStrings.stack,
+        middle: p3UROutputStrings.middle,
+      },
+    },
+    {
+      id: 'FRU P3 Ultimate Rel 3nd Bait',
+      type: 'GainsEffect',
+      netRegex: { effectId: '9A0' },
+      condition: (data, matches) => data.phase === 'p3-ur' && data.me === matches.target,
+      delaySeconds: 31,
+      alertText: (data, _matches, output) => {
+        const debuff = data.p3RelativityDebuff;
+        if (debuff === undefined)
+          return;
+
+        if (data.triggerSetConfig.ultimateRel !== 'yNorthDPSEast')
+          return output[debuff]!();
+
+        const dirStr = data.p3RelativityMyDirStr;
+
+        if (debuff !== 'mediumFire')
+          return output.yNorthStrat!({ debuff: output[debuff]!(), dir: output.middle!() });
+        return output.yNorthStrat!({ debuff: output.mediumFire!(), dir: dirStr });
+      },
+      outputStrings: {
+        yNorthStrat: p3UROutputStrings.yNorthStrat,
+        shortFire: p3UROutputStrings.avoidStoplights,
+        mediumFire: p3UROutputStrings.baitStoplight,
+        longFire: p3UROutputStrings.avoidStoplights,
+        ice: p3UROutputStrings.avoidStoplights,
+        middle: p3UROutputStrings.middle,
+        out: p3UROutputStrings.out,
+      },
+    },
+    {
+      id: 'FRU P3 Ultimate Rel Rewind',
+      type: 'GainsEffect',
+      netRegex: { effectId: '994' }, // 994 - Return
+      condition: (data, matches) => data.phase === 'p3-ur' && data.me === matches.target,
+      delaySeconds: (_data, matches) => parseFloat(matches.duration) - 4,
+      countdownSeconds: 4,
+      response: Responses.lookAway('alarm'),
+    },
+    {
+      id: 'FRU P3 Shockwave Pulsar',
+      type: 'StartsUsing',
+      netRegex: { id: '9D5A', source: 'Oracle of Darkness', capture: false },
+      response: Responses.bigAoe(),
+    },
+    {
+      id: 'FRU P3 Black Halo',
+      type: 'StartsUsing',
+      netRegex: { id: '9D62', source: 'Oracle of Darkness' },
+      response: Responses.sharedTankBuster(),
+    },
```

Comment:
Black Halo is a cone, might be good to indicate that here? I only suggest it because it's very easy to lose track of position after the rewinds go off. I'm fine with leaving it as-is, too.

Reply:
I would *strongly* recommend doing a custom "shared cleave" of some sort, since this indicator usually isn't used for conal busters.